app/models.py

```
@@ -2,21 +2,26 @@
+    Parses structurally rather than string-sniffing: the old check let
+    hostless URLs like ``https:///room`` through (``/room`` contains a ``/``),
+    which then flowed to MeetingBaas as a malformed join.
+    """
     if not value:
         raise ValueError("meeting_url is required")
 
     normalized = value.strip()
-    if not normalized.startswith(("http://", "https://")):
+    parsed = urlparse(normalized)
+    if parsed.scheme not in ("http", "https"):
         raise ValueError("meeting_url must start with http:// or https://")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)